camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java

```diff
@@ -0,0 +1,184 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.builder;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.model.DataFormatDefinition;
+import org.apache.camel.model.transformer.CustomTransformerDefinition;
+import org.apache.camel.model.transformer.DataFormatTransformerDefinition;
+import org.apache.camel.model.transformer.EndpointTransformerDefinition;
+import org.apache.camel.model.transformer.TransformerDefinition;
+import org.apache.camel.spi.Transformer;
+
+/**
+ * A <a href="http://camel.apache.org/dsl.html">Java DSL</a> which is
+ * used to build a {@link org.apache.camel.spi.Transformer} and register into {@link org.apache.camel.CamelContext}.
+ */
+public class TransformerBuilder {
+
+    private String scheme;
+    private String from;
+    private String to;
+    private String uri;
+    private DataFormatDefinition dataFormat;
+    private Class<? extends Transformer> clazz;
+    private String beanRef;
+
+    /**
+     * Set a scheme name supported by the transformer.
+     *
+     * @param scheme scheme name
+     */
+    public TransformerBuilder scheme(String scheme) {
+        this.scheme = scheme;
+        return this;
+    }
+
+    /**
+     * Set the 'from' data type .
+     *
+     * @param from 'from' data type
+     */
+    public TransformerBuilder from(String from) {
+        this.from = from;
+        return this;
+    }
+
+    /**
+     * Set the 'from' data type using Java class.
+     *
+     * @param clazz 'from' Java class
+     */
+    public TransformerBuilder from(Class<?> from) {
+        this.from = "java:" + from.getName();
+        return this;
+    }
+
+    /**
+     * Set the 'to' data type .
+     *
+     * @param to 'to' data type
+     */
+    public TransformerBuilder to(String to) {
+        this.to = to;
+        return this;
+    }
+
+    /**
+     * Set the 'to' data type using Java class.
+     *
+     * @param clazz 'to' Java class
+     */
+    public TransformerBuilder to(Class<?> to) {
+        this.to = "java:" + to.getName();
+        return this;
+    }
+
+    /**
+     * Set the URI to be used for the endpoint {@code Transformer}.
+     * @see {@code EndpointTransformerDefinition}, {@code ProcessorTransformer}
+     * 
+     * @param uri endpoint URI
+     */
+    public TransformerBuilder withUri(String uri) {
+        resetType();
+        this.uri = uri;
+        return this;
+    }
+
+    /**
+     * Set the {@code DataFormatDefinition} to be used for the {@code DataFormat} {@code Transformer}.
+     * @see {@code DataFormatTransformerDefinition}, {@code DataFormatTransformer}
+     * 
+     * @param dfd {@code DataFormatDefinition}
+     */
+    public TransformerBuilder withDataFormat(DataFormatDefinition dfd) {
```

Comment:
optional: what do you think to use a more meaningful parameter name such as dataFormatDefinition, it is longer but easier to read from my point of view